MailerQ/ApiClient/ApiClient.cs

```diff
@@ -15,186 +17,200 @@ public class ApiClient : IApiClient
+        public ApiClient(
+            ILogger<ApiClient> logger,
+            IOptions<ApiClientSettings> options,
+            IHttpClientFactory httpFactory) : this(logger, options.Value, httpFactory) { }
 
-        public bool Post(Error error)
+        public async Task<bool> Post(Error error)
         {
             try
             {
                 var jsonContent = JsonConvert.SerializeObject(error);
-                var result = MailerQApiClient(EndpointError, MethodPost, jsonContent);
+                var result = await MailerQApiClientPostAsync(error.Endpoint, jsonContent);
                 return result == HttpStatusCode.OK;
             }
             catch (Exception exception)
             {
                 _logger.LogError(exception,
                     "Endpoint: {Endpoint}. Method: {Method}.",
-                    EndpointError,
-                    MethodPost);
+                    error.Endpoint,
+                    nameof(Post));
                 throw exception;
             }
         }
 
-        public bool Post(Pause pause)
+        public async Task<bool> Post(Pause pause)
         {
             try
             {
                 var jsonContent = JsonConvert.SerializeObject(pause);
-                var result = MailerQApiClient(EndpointPause, MethodPost, jsonContent);
+                var result = await MailerQApiClientPostAsync(pause.Endpoint, jsonContent);
                 return result == HttpStatusCode.OK;
             }
             catch (Exception exception)
             {
                 _logger.LogError(exception,
                     "Endpoint: {Endpoint}. Method: {Method}.",
-                    EndpointPause,
-                    MethodPost);
+                    pause.Endpoint,
+                    nameof(Post));
                 throw exception;
             }
         }
 
-        public bool Post(Inject inject)
+        public async Task<bool> Post(Inject inject)
         {
             try
             {
                 var jsonContent = JsonConvert.SerializeObject(inject);
-                var result = MailerQApiClient(EndpointInject, MethodPost, jsonContent);
+                var result = await MailerQApiClientPostAsync(inject.Endpoint, jsonContent);
                 return result == HttpStatusCode.OK;
             }
             catch (Exception exception)
             {
                 _logger.LogError(exception,
                     "Endpoint: {Endpoint}. Method: {Method}.",
-                    EndpointInject,
-                    MethodPost);
+                    inject.Endpoint,
+                    nameof(Post));
                 throw exception;
             }
         }
-        public bool Delete(Error error)
+
+        private async Task<HttpStatusCode> MailerQApiClientPostAsync(string endpoint, string content)
+        {
+            _apiUri.Path = Path.Combine(_apiUri.Path, endpoint);
```

Comment:
Mutate an object field should not look as a good idea, it would limit thread safety (maybe you do not need it now, but you do not know about the future).

Do you think that it is possible remove this state mutation as auxiliar step in a method?

(Reading this, I understand that I am not being so clear to explain myself, if it is not clear, please call me)

Reply:
I understand that is not a good idea change  `_apiUri.Path` here, I am right? 
If it is that, I think that it is possible to move `Path.Combine(_apiUri.Path, endpoint)` to a place where is used `_apiUri.Path`, maybe.

Reply:
You said it much better than me :)